### ISSUE TITLE
ImageFolder Dataset: `make_dataset` with the same (sorted) order.

### ISSUE THREAD

**Issue by jinjiren** [pytorch/vision#192]
When creating the dataset using `ImageFolder`, different machines can have different orders of images. 

This may not be an issue for the training set, but when I want to compare the validation results among different machines (especially when I want to ensemble the softmax outputs of validation set), different orders of images can be annoying.

A simple modification can solve the problem: currently the `make_dataset` function uses `for target in os.listdir(dir):`, we can simply change it to `for target in sorted(os.listdir(dir)):`.

Since the loading procedure can be shuffled, I have not found creating the dataset with a sorted order has any drawbacks. Actually, it will be much easier for the users when they want to ensemble the outputs of the full dataset among different machines (where `os.listdir(dir)` might produce different orders of results`).

**Comment by jinjiren** [comment]
Also, it is difficult to reproduce the results on different machines. Even setting random seed will not produce the same ordered shuffle in training because the original order of data are different.

**Comment by soumith** [comment]
merged and fixed. thanks!
